### PR TITLE
Metamask Connection

### PR DESCRIPTION
**Pull request by joaop21**
This PR adds a button on NavBar, which connects a wallet.
The user can change to different accounts inside the same wallet, or exit the wallet.
The button only has 2 states: 
- "Connect Wallet" - When there is no wallet connected to the site.
- "Wallet Connected" - When there is a wallet connection.

### CODE REVIEW

File: web/src/components/Header.js [hunk 1]
```diff
@@ -32,6 +34,23 @@ const Header = (props) => {
     </Grid>
   );
 
+  WalletService.changeAccounts(setWallet);
+
+  const connect_button =
```
Comment (Bishop19) [inline web/src/components/Header.js]: Show only when logged in?

Comment (joaop21) [reply web/src/components/Header.js]: With the current approach you can connect the wallet at any time, but I can change it to when the user is logged in.

**Comment by pferreira101** [comment]
This looks nice and seems to be working. I agree with Pedro, perhaps it would be better to only show the Connect button if the user is logged in.
On Brave I can connect with the wallet easily, yet on Chrome and Safari the page doesn't even load. It seems that some authorization or extension is missing 🤔 Anyone else having the same problem?

**Comment by Bishop19** [comment]
> On Brave I can connect with the wallet easily, yet on Chrome and Safari the page doesn't even load. It seems that some authorization or extension is missing  Anyone else having the same problem?

I used Chrome, all good

**Comment by joaop21** [comment]
> This looks nice and seems to be working. I agree with Pedro, perhaps it would be better to only show the Connect button if the user is logged in.
> On Brave I can connect with the wallet easily, yet on Chrome and Safari the page doesn't even load. It seems that some authorization or extension is missing 🤔 Anyone else having the same problem?

I used Brave and Firefox, and all worked just fine

File: web/src/components/NavMenu.js [hunk 2]
```diff
@@ -1,17 +1,36 @@
...
+      <Button
+        variant="contained"
+        color="secondary"
+        onClick={() => WalletService.connectToWallet()}
+      >
+        Connect Wallet
+      </Button>
+    ) : (
+      <Button variant="contained" color="secondary" disabled>
+        Wallet Connected
```
Comment (Bishop19) [inline web/src/components/NavMenu.js]: I think the text is too big, maybe just "Connected"?

**Review by Bishop19 — APPROVED** [review]

File: web/src/components/NavMenu.js (hunk 2 above)
Comment (joaop21) [reply web/src/components/NavMenu.js]: Done

**Review by Bishop19 — APPROVED** [review]

**Review by pferreira101 — APPROVED** [review]
🚀

**Pull request merged**